x-pack/plugins/ml/public/embeddables/single_metric_viewer/embeddable_single_metric_viewer_container.tsx

```diff
@@ -81,13 +73,25 @@ export const EmbeddableSingleMetricViewerContainer: FC<
     onRenderComplete
   );
   const selectedJobId = data?.jobIds[0];
+  // Need to make sure we fall back to `undefined` if `functionDescription` is an empty string,
+  // otherwise anomaly table data will not be loaded.
+  const functionDescription =
+    (data?.functionDescription ?? '') === '' ? undefined : data.functionDescription;
   const previousRefresh = usePrevious(lastRefresh ?? 0);
   const mlTimeSeriesExplorer = useTimeSeriesExplorerService();
 
   // Holds the container height for previously fetched data
   const containerHeightRef = useRef<number>();
   const toastNotificationService = useToastNotificationService();
 
+  useEffect(function setUpJobsLoaded() {
```

Comment:
Again, would prefer not to reintroduce the dependency on a service we're wanting to completely refactor or replace - as we've got access here to ml services we should be able to make the requests and such that we need - perhaps in a custom hook or separate, updated service as mentioned before. 

cc @jgowdyelastic  for thoughts

Reply:
The `LinksMenu` component for the anomaly marker popover still uses `mlJobService` and I'd like to avoid to refactor that file further as part of this PR; because it's still the same setup on the Anomaly Explorer and Single Metric Viewer it felt like the path with less churn for this PR to still use `mlJobService`. Suggest to get rid of `mlJobService` as a whole in follow ups.

Reply:
`mlJobService` has been on my list for complete removal for a while. Most of it isn't used anymore, and what is used should be refactored away.